src/config/frontend/EdFi.DmsConfigurationService.Frontend.AspNetCore.Tests.Unit/Modules/ApplicationModuleTests.cs

```diff
@@ -0,0 +1,381 @@
+                        {
+                            Id = 1,
+                            ApplicationName = "Test Application",
+                            ClaimSetName = "ClaimSet",
+                            VendorId = 1,
+                            ApplicationEducationOrganizations = [1]
+                        }
+                    )
+                );
+
+            A.CallTo(() => _repository.UpdateAsync(A<Application>.Ignored))
+                .Returns(new UpdateResult.UpdateSuccess(1));
+
+            A.CallTo(() => _repository.DeleteAsync(A<long>.Ignored))
+                .Returns(new DeleteResult.DeleteSuccess(1));
+        }
+
+        [Test]
+        public async Task Should_return_success_response()
+        {
+            // Arrange
+            using var client = SetUpClient();
+
+            var addResponse = await client.PostAsync(
+                "/v2/applications",
+                new StringContent(
+                    """
+                    {
+                      "ApplicationName": "Application 11",
+                      "ClaimSetName": "Test",
+                      "VendorId": 1,
+                      "ApplicationEducationOrganizations": [1]
+                    }
+                    """,
+                    Encoding.UTF8,
+                    "application/json"
+                )
+            );
+
+            var getResponse = await client.GetAsync("/v2/applications");
+            var getByIdResponse = await client.GetAsync("/v2/applications/1");
+            var updateResponse = await client.PutAsync(
+                "/v2/applications/1",
+                new StringContent(
+                    """
+                    {
+                       "id": 1,
+                       "ApplicationName": "Application 11",
+                        "ClaimSetName": "Test",
+                        "VendorId": 1,
+                        "ApplicationEducationOrganizations": [1]
+                    }
+                    """,
+                    Encoding.UTF8,
+                    "application/json"
+                )
+            );
+            var deleteResponse = await client.DeleteAsync("/v2/applications/1");
+
+            //Assert
+            addResponse.StatusCode.Should().Be(HttpStatusCode.Created);
+            getResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+            getByIdResponse.StatusCode.Should().Be(HttpStatusCode.OK);
+            updateResponse.StatusCode.Should().Be(HttpStatusCode.NoContent);
+            deleteResponse.StatusCode.Should().Be(HttpStatusCode.NoContent);
+        }
+    }
+
+    [TestFixture]
+    public class FailureValidationTests : ApplicationModuleTests
+    {
+        [Test]
+        public async Task Should_return_bad_request()
+        {
+            // Arrange
+            using var client = SetUpClient();
+
+            string invalidBody = """
+                {
+                   "ApplicationName": "Application 101",
```

Comment:
It might slightly improve coverage metrics to trip validations on all the properties that have a rule. For example ApplicationName with a length greater than 256. See vendor test for example.